docs/DeveloperGuide.md

```diff
@@ -1,34 +1,1040 @@
+* In the `findRating()` method, if the user input such as the book to search for is missing, a message will be printed
+to inform the user and the method is returned.
+* The list of ratings will be looped to see if the rating exists for the particular book.
+* Since the ratings of book is unique, the loop will be broken when a rating is found and details of the rating
+will be printed to the user.
+
+## Appendix: Requirements
+
+### Product scope
+
+#### Target user profile
+The intended user of Quotesify is someone that meets the following criterias:
+* reads a lot
+* has difficulty remembering content after reading them
+* can type fast
+* prefers typing over other means of input
+* prefers using desktop applications
+* comfortable with using Command Line Interface (CLI) applications
+
+#### Value proposition
+Quotesify will help you to improve your reading experience with quick and easy features such as book management,
+quote management, progress tracker, category management and a rating system for your books.
+
+#### User Stories
+
+|Version| As a ... | I want to ... | So that I ...|
+|--------|----------|---------------|------------------|
+|v1.0|reader|enter good quotes and phrases from a book I read|can quickly refer back to it at any time|
+|v1.0|user|categorize my listings|can view quotes from a specific category that I want|
+|v1.0|long time user|have a page to see a list of books and categories of my notes|can navigate into the relevant book/category without having to remember the titles|
+|v1.0|user|give rating for the books I read|can recommend book titles to others when asked|
+|v1.0|beginner reader|add deadlines to books I am reading|can keep track of my readling deadlines|
+|v1.0|avid reader|be able to keep track of books|can filter out books I have read and those that are on my list of books to read|
+|v1.0|forgetful user|add bookmarks to my book|can find the page where I last stopped|
+|v1.0|user|categorise my books or quotes|can view items from a specific category whenever I need|
+|v1.0|user|save quotes I find meaningful|can view my favourite quotes whenever I want|
+|v2.0|forgetful reader|be reminded of quotes I saved|can remember them better in the long run|
+|v2.0|student and reader|pen my thoughts to a highlighted quote or text|can expand on certain ideas or express how I feel|
+|v2.0|long time user|be able to search for keywords|can find specific quotes I want from the list|
+|v2.0|user after some time|find a book rating by its book title|do not have to go through the whole list|
+
+#### Non-Functional Requirements
+1. Should work on major Operating Systems (OS) such as Windows and Mac with at least `Java 11` installed.
+2. A user should have no problems using the various commands without referring to the help page after some time.
+
+#### Glossary
 * *glossary item* - Definition
 
-## Instructions for manual testing
+## Appendix: Instructions for manual testing
```

Comment:
can every section in this appendix be partitioned based on functionality to increase readability and tidiness/neatness such that the next developer would know where to find? For example, you can try segment 'List' functionality with every other list functionality.

Reply:
![image](https://user-images.githubusercontent.com/57656916/97530887-0020ea00-19ee-11eb-83ba-0f338ffc121d.png)
Can try to group these two launches as one to make easier find?